lib/src/main/java/tech/relaycorp/relaydroid/messaging/ReceiveMessages.kt

```diff
@@ -0,0 +1,41 @@
+package tech.relaycorp.relaydroid.messaging
+
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaydroid.Relaynet
+import tech.relaycorp.relaydroid.Storage
+import tech.relaycorp.relaynet.bindings.pdc.PDCClient
+import tech.relaycorp.relaynet.bindings.pdc.Signer
+import tech.relaycorp.relaynet.bindings.pdc.StreamingMode
+import tech.relaycorp.relaynet.messages.Parcel
+
+internal class ReceiveMessages(
+    private val pdcClientBuilder: () -> PDCClient = { PoWebClient.initLocal(Relaynet.POWEB_PORT) }
+) {
+
+    fun receive() =
+        getNonceSigners()
+            .flatMapLatest { nonceSigners ->
+                pdcClientBuilder().use {
+                    it.collectParcels(nonceSigners, StreamingMode.CloseUponCompletion)
+                        .map { parcelCollection ->
+                            val parcel = Parcel.deserialize(parcelCollection.parcelSerialized)
```

Comment:
It's safer to use the following instead:

```
val parcel = parcelCollection.deserializeAndValidateParcel()
```

... Because it also validates the parcel and checks authorisation.

On the other hand, how should we handle `RAMFException`s and `InvalidMessageException`s here? If I'm reading this right, a single invalid parcel can lead to a deadlock because the gateway will continue to send it until we ACK it. For this reason, I think we should ACK the collection of an invalid parcel and log an error; in the future we might want to allow devs to override this behaviour. Thoughts?

Reply:
Yeah, we can just log the errors for now, ack, and skip delivering the messages to the app.